File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <iostream>
using namespace std;

int gcd(int a, int b) {
    if(a%b == 0) {
        return b;
    }
    else {
        return gcd(b, a%b);
    }
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int a, b, gcf, lcm;
    cin>>a>>b;

    if(a>b) gcf = gcd(a, b);
    else gcf = gcd(b, a);

    lcm = (a*b) / gcf;

    cout<<gcf<<"\n"<<lcm<<"\n";

    return 0;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
const gcd = (n, m) => {
    let d = n;
    
    while (d > 0) {
        if ((n % d === 0) && (m % d === 0)) {
            return d;
        }
        d--;
    }
}

const lcm = (n, m) => {
    let d = 1;
    
    while (d <= n) {
        if ((m * d) % n === 0) {
            return m * d;
        }
        d++;
    }
}

function solution(n, m) {
    let big = 0;
    let small = 0;
    
    if (n > m) {
        big = n;
        small = m;
    } else {
        big = m;
        small = n;
    }
    
    return [gcd(small, big), lcm(small, big)];
}